integration-tests/src/test_loop/utils/setups.rs

```diff
@@ -57,3 +60,33 @@ pub fn standard_setup_1() -> TestLoopEnv {
         .clients(clients)
         .build()
 }
+
+pub fn derive_new_epoch_config_from_boundary(
+    base_epoch_config: &EpochConfig,
+    boundary_account: &AccountId,
+) -> EpochConfig {
+    let base_shard_layout = &base_epoch_config.shard_layout;
+    let mut epoch_config = base_epoch_config.clone();
+    epoch_config.shard_layout =
+        ShardLayout::derive_shard_layout(&base_shard_layout, boundary_account.clone());
+    tracing::info!(target: "test", ?base_shard_layout, new_shard_layout=?epoch_config.shard_layout, "shard layout");
+    epoch_config
+}
+
+pub fn two_upgrades_voting_schedule(
```

Comment:
Maybe add a comment saying that the two protocol upgrades will be applied in two successive epochs

Reply:
fixed in https://github.com/near/nearcore/pull/12762/commits/f3accf0db2b5f5ded7e44a9229429aef92a3150a